### DISCUSSION TITLE
Utils.copy() does not maintain the new instance metadata

### DISCUSSION THREAD

**Discussion by ssljivic** [mikro-orm/mikro-orm#5575] [General]
### Describe the bug

After making a copy of the managed entity using `Utils.copy()`, the new entity instance is not managed. When changes made to the new instance are persisted, MikroOrm will produce an INSERT statement instead of an UPDATE statement and due to duplicate PK constraint this INSERT will fail.

New copy/instance is not managed and this is probably the reason INSERT is generated. 
Is there a way to "make" the copied instance as managed without triggering any additional SQL statement against the DB?

### Reproduction

```ts
import { Entity, PrimaryKey, Property, Utils } from '@mikro-orm/core'
import { EntityManager, MikroORM, PostgreSqlDriver } from '@mikro-orm/postgresql'

@Entity()
export class Test {
  @PrimaryKey()
  name: string

  @Property()
  description: string
}

describe('MikroORM', () => {
  let orm: MikroORM
  let em: EntityManager

  beforeAll(async () => {
    orm = await MikroORM.init({
      host: 'localhost',
      port: 5432,
      user: 'mikro-orm-test',
      password: 'mikro-orm-test',
      dbName: 'mikro-orm-test',

      driver: PostgreSqlDriver,

      debug: true,
      entities: [Test],
    })
    await orm.schema.refreshDatabase({ wrap: true })

    em = orm.em.fork()
  })

  afterAll(async () => {
    await orm.close(true)
  })

  it('Utils.copy() should copy state of the entity instance', async () => {
    const t = new Test()
    t.name = 'foo'
    t.description = 'bar'

    await em.persistAndFlush(t)

    const t2 = Utils.copy(t)

    // I would expect that the next em.persist() operation creates UPDATE and not INSERT SQL statement.
    t2.description = 'Bar'
    await em.persistAndFlush(t2)
  })
})
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.5

### Node.js version

18.17.1

### Operating system

MacOS 14.0

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Why are you expecting this to work? `copy` makes a copy, that should be pretty clear from the name. A copy is a copy, not the managed entity instance. By copying you are not merging anything, you just... copy :]

This is a wontfix to me, you need to call `em.merge()` on the copy if you want it to be managed.

**Reply by ssljivic** [reply]
I would be ok with merge, but it does not work as well:

```ts
    const t2 = Utils.copy(t)
    em.merge(t2)

    // I would expect that the next em.persist() operation creates UPDATE and not INSERT SQL statement.
    t2.description = 'Bar'
    await em.persistAndFlush(t2)
```

Even after merging the copy, the entity manager produces INSERT, instead of UPDATE.

**Reply by ssljivic** [reply]
OK, just tried `em.merge(t2, { refresh: true })` and that seems to work.

@B4nan can you please provide a little explanation what is the `refresh: true` option? I expected that it would trigger some SQL statements, but it did not.

**Reply by B4nan** [reply]
Sync methods will never trigger any SQL statements, that's granted. 

IIRC in `em.merge` the refresh flag controls whether you replace the entity instance in the identity map or you just merge the values into it and keep the object reference in place (but maybe things changed over time, I would need to give it a try first).
